### PR TITLE
Add GCP auto-discovery docs

### PR DESCRIPTION
**Pull request by atburke**
This PR adds documentation for GCP VM auto-discovery (#28562).

### CODE REVIEW

File: docs/pages/includes/discovery/discovery-config.yaml [hunk 1]
```diff
@@ -54,4 +55,23 @@ discovery_service:
       resource_groups: ["group1", "group2"]
       # Azure resource tag filters used to match resources.
       tags:
-        "*": "*"
\ No newline at end of file
+        "*": "*"
+    # Matchers for discovering GCP-hosted resources.
+    gcp:
+      # GCP resource types. Valid options are:
+      # 'gke' - discovers and registers GCP kubernetes clusters.
```
Comment (zmb3) [inline docs/pages/includes/discovery/discovery-config.yaml:62]:
```suggestion
      # 'gke' - discovers and registers GKE kubernetes clusters.
```

**Review by zmb3 — APPROVED** [review]

File: docs/pages/server-access/guides/gcp-discovery.mdx [hunk 2]
```diff
@@ -0,0 +1,238 @@
+---
+title: Automatically Discover GCP Compute Instances (Preview)
```
Comment (r0mant) [inline docs/pages/server-access/guides/gcp-discovery.mdx:2]: We're revising our Preview policy and no longer going to call every single thing we release as preview. Let's remove this.
```suggestion
title: Automatically Discover GCP Compute Instances
```

File: docs/pages/server-access/guides/gcp-discovery.mdx [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+host keys. Enable guest attributes by setting `enable-guest-attributes` to
+`TRUE` in the instance's metadata.
+
+If guest attributes are enabled during instance creation, the guest attributes
+will automatically be populated with the instance's host keys. If guest
+attributes were enabled after the instance was created, you can manually add
+the host keys to the guest attributes by running the following script on the instance:
+
+```sh
+#!/usr/bin/env bash
```
Comment (r0mant) [inline docs/pages/server-access/guides/gcp-discovery.mdx:148]: Is this something that can be done via GCP API or with `gcloud` instead?

**Review by ptgott — APPROVED** [review]

**Comment by ptgott** [comment]
@atburke This looks good to me, though I'd double-check with @r0mant before we merge

**Review by r0mant — APPROVED** [review]

**Pull request merged**